mne/time_frequency/tfr.py

```diff
@@ -1012,6 +1017,82 @@ def save(self, fname, overwrite=False, *, verbose=None):
         """
         write_tfrs(fname, self, overwrite=overwrite)
 
+    @fill_doc
+    def to_data_frame(self, picks=None, index=None, long_format=False,
+                      time_format='ms'):
+        """Export data in tabular structure as a pandas DataFrame.
+
+        Channels are converted to columns in the DataFrame. By default,
+        additional columns ``'time'``, ``'freq'``, ``'epoch'``, and
+        ``'condition'`` (epoch event description) are added, unless ``index``
+        is not ``None`` (in which case the columns specified in ``index`` will
+        be used to form the DataFrame's index instead). ``'epoch'``, and
+        ``'condition'`` are not supported for ``AverageTFR``.
+
+        Parameters
+        ----------
+        %(picks_all)s
+        %(df_index_epo)s
+            Valid string values are ``'time'``, ``'freq'``, ``'epoch'``, and
+            ``'condition'`` for ``EpochsTFR`` and ``'time'`` and ``'freq'``
+            for ``AverageTFR``.
+            Defaults to ``None``.
+        %(df_longform_epo)s
+        %(df_time_format)s
+
+            .. versionadded:: 0.23
+
+        Returns
+        -------
+        %(df_return)s
+        """
+        # check pandas once here, instead of in each private utils function
+        pd = _check_pandas_installed()  # noqa
+        # arg checking
+        if isinstance(self, EpochsTFR):
+            valid_index_args = ['time', 'freq', 'epoch', 'condition']
+        else:
+            valid_index_args = ['time', 'freq']
```

Comment:
(nitpick) this is slightly more DRY

```suggestion
        valid_index_args = ['time', 'freq']
        if isinstance(self, EpochsTFR):
            valid_index_args.append(['epoch', 'condition'])
```